### PR TITLE
Add network requests allowed API

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Has at least one reviewer tested the PR?

## Description

Add API to block network requests.

## Related PRs or issues

[AB#85220](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/85220)

### CODE REVIEW

File: Assets/AppCenter/Plugins/iOS/Core/AppCenterUnity.mm [hunk 1]
```diff
@@ -44,6 +44,17 @@ void appcenter_unity_set_enabled(bool isEnabled)
...
 
+void appcenter_unity_set_network_requests_allowed(bool isAllowed)
+{
+  [MSACAppCenter setNetworkRequestsAllowed:isAllowed];
+}
+
+bool appcenter_unity_is_network_requests_allowed()
+{
+  // This is a workaround to convert the BOOL value.
+  return [MSACAppCenter isNetworkRequestsAllowed] ? [@"YES" boolValue] : [@"NO" boolValue];
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/iOS/Core/AppCenterUnity.mm]: Ohh....

```suggestion
  return (bool)[MSACAppCenter isNetworkRequestsAllowed];
```

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

**Review by Jamminroot — DISMISSED** [review]

File: Assets/AppCenter/AppCenterSettings.cs [hunk 2]
```diff
@@ -41,6 +41,9 @@ public class AppCenterSettings : ScriptableObject
 
     public LogLevel InitialLogLevel = LogLevel.Info;
 
+    [Tooltip("By default, the network requests is allowed. Uncheck this to check for updates manually instead.")]
```
Comment (MatkovIvan) [inline Assets/AppCenter/AppCenterSettings.cs]: > Uncheck this to check for updates manually instead.

This does not look related. Need to proper wording

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
 # App Center SDK for Unity Change Log
 
+## Version 4.2.0  (Under development)
```
Comment (MatkovIvan) [inline CHANGELOG.md]:
```suggestion
## Version 4.2.0 (Under development)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
+
+### AppCenter
+
+* **[Feature]** Add a `IsNetworkRequestsAllowed` API to block network requests.
```
Comment (MatkovIvan) [inline CHANGELOG.md]: Harmonize wording with other platforms

**Review by MatkovIvan — APPROVED** [review]
